### PR TITLE
Feat/add event bottomsheet integration

### PR DESCRIPTION
**Pull request by ItsaBox9368**
**Summary**
This PR modifies the original AddEventPopUp (now AddEventScreen) and integrates it into the BottomSheet in the MapScreen in order to achieve a cleaner representation, whilst staying on theme with the rest of the project, especially memory creation.

**Key Features**
- Improved UI of AddEventScreen
- New Enum class BottomSheetScreen for BottomSheetContent navigation, making it more modular and scalable
- Added functions for BottomSheetScreen switching in MapScreenViewModel. especially for AddEventScreen to work in Bottomsheet
- More user-friendly UI behaviour
- Tests for added functions in MapScreenViewModel

**Testing**
- New tests for new functions in MapScreenViewModel
- Removed tests where logic was changed and thus made the tests irrelevant

**Preview**
image

## Summary by CodeRabbit

* **New Features**
  * Add Event screen in the bottom sheet with title, date & time pickers, location, description, tags, public/private toggle, and save/cancel actions

* **UI Improvements**
  * Event creation flows integrated as a distinct bottom-sheet screen with clearer labels and improved location input behavior; updated date/time wording

* **Tests**
  * New and updated unit and UI tests for bottom-sheet navigation and Add Event workflows

* **Chores**
  * Package and resource reorganization; added string resources

### CODE REVIEW

**Review by Acheline — CHANGES_REQUESTED** [review]
## Overall review

This is a substantial refactor that migrates event creation from a dialog to a bottom-sheet AddEventScreen and introduces a clear screen state via BottomSheetScreen. The direction is solid and aligns with the app’s UX. That said, there are a few compile blockers, logic bugs, insufficient tests, and resource issues that should be addressed before merging.

Line coverage is currently under 80%, which is sadly not sufficient for merging. Some aspects may require additional tests, refactoring, and documentation improvements.

### High-priority findings (blockers)

1. Compile errors with Material3 colors :
ButtonColors(...) is not publicly constructible. Use ButtonDefaults.buttonColors(...).

2. Broken validation logic :
Missing || in the error expression and date/time validity not enforced when fields are blank.

3. Invalid strings.xml :
Trailing > characters and duplicate keys (e.g., date_field) will break the build.

4. Tests still reference old test tags and packages :
Android tests use both AddEventPopUpTestTags and AddEventScreenTestTags in chained calls (won’t find any nodes). Some test packages still reference .AddEvent vs .event.

5. modifier param ignored in AddEventScreen root layout, which blocks parent sizing/test semantics.

### Depth areas (what to add / tighten)
**Testing & Edge Cases**

Add instrumented/unit tests for:

- Date/time required-field validation (save disabled when either is blank).

- saveEvent happy path and failure path (exceptions from repo).

- MapScreenViewModel transitions: showAddEventForm(), onAddEventCancel(), restoring previous sheet size/state.

- System pickers: provide injectable lambdas for setting date/time to make tests deterministic (no system dialogs).

- Tag validation edge cases (#tag, separator variants, invalid chars).

- Location search debounce and error cases (no matches, network errors).

**Error Handling & User Feedback**

- Centralize user-visible errors in string resources or constants.

- Show a transient snackbar/toast when save fails (instead of silent failure).

**Naming & Readability**

- Align resource strings: “New Event” vs “New Event Manager”.

- Prefer positively phrased names for booleans (isPublic, isEventValid already good; ensure consistency elsewhere).

**Documentation**

Add brief KDoc at file top of AddEventScreen.kt and BottomSheetContent.kt describing:

- How date/time selection flows into a single Timestamp.

- What minimum fields are required and how validation is composed.

- How BottomSheetScreen coordinates screen swapping.

**UX & Accessibility**

- Ensure action Icons (Close, Check) have accurate contentDescription.

- Consider showing the “fields missing/incorrect” banner only after first submit attempt.

**Performance / Robustness**

- Replace repeated SimpleDateFormat parsing with a single type-safe model (store hour/minute ints; only format for display).

- Consider debouncing locationViewModel.onQueryChanged to avoid chatty requests.

File: app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+  val gotLocation = remember {
+    mutableStateOf(Location(location.value, LATITUDE_DEFAULT, LONGITUDE_DEFAULT))
+  }
+  val locations by locationViewModel.locations.collectAsState()
+
+  val error =
+      titleError.value || title.value.isBlank() ||
+          descriptionError.value || description.value.isBlank() ||
+          locationError.value || location.value.isBlank()
+          timeError.value ||
```
Comment (Acheline) [inline app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt]: - Error : There’s a missing || before timeError.value, and date/time blanks aren’t considered, so save can appear enabled prematurely.
- Suggestion : 
``
val error =
titleError.value || title.value.isBlank() ||
descriptionError.value || description.value.isBlank() ||
locationError.value || location.value.isBlank() ||
date.value.isBlank() || dateError.value ||
time.value.isBlank() || timeError.value
``

File: app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+          if (locationError.value) stringResource(R.string.location_field) else null,
+          if (descriptionError.value) stringResource(R.string.description_field) else null,
+          if (tagError.value) stringResource(R.string.tag_text) else null,
+          if (timeError.value) stringResource(R.string.time) else null)
+
+  val isEventValid = !error && isLoggedIn.value
+
+  val scrollState = rememberScrollState()
+
+  Column(modifier = Modifier.fillMaxWidth().verticalScroll(scrollState)) {
```
Comment (Acheline) [inline app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt]: - Nitpick : The root container ignores modifier, which prevents parents from adding test tags/layout.
- Suggestion : 
``
Column(modifier = modifier.fillMaxWidth().verticalScroll(scrollState)) {
``

File: app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+  const val INPUT_EVENT_LOCATION = "inputEventLocation"
+  const val EVENT_CANCEL = "eventCancel"
+  const val EVENT_SAVE = "eventSave"
+  const val ERROR_MESSAGE = "errorMessage"
+
+  const val PUBLIC_SWITCH = "publicSwitch"
+
+  const val PUBLIC_TEXT = "publicText"
+
+  const val SCREEN = "AddEventPopUp"
```
Comment (Acheline) [inline app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt]: - Nitpick : SCREEN references the old name.
- Suggestion : 
``
const val SCREEN = "AddEventScreen"
``

File: app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+          onTimeClick()
+        } else {
+          val calendar = Calendar.getInstance()
+          val hour = calendar[Calendar.HOUR_OF_DAY]
+          val minute = calendar[Calendar.MINUTE]
+
+          TimePickerDialog(
+                  context,
+                  { _, pickedHour, pickedMinute ->
+                    selectedTime.value =
```
Comment (Acheline) [inline app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt]: - Nitpick : "HHhmm" with a .replace("h","") is fragile. Prefer plain "HHmm" for internal state and just display "HH'h'mm".
- Suggestion : 
``
selectedTime.value = pickedHour.toString().padStart(2, '0') + pickedMinute.toString().padStart(2, '0') // internal "HHmm"
``

File: app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt [hunk 5]
```diff
@@ -0,0 +1,475 @@
...
+      },
+      modifier = Modifier.width(200.dp).testTag(AddEventScreenTestTags.PICK_EVENT_TIME),
+      shape = RoundedCornerShape(4.dp),
+      colors =
+          ButtonColors(
+              containerColor = MaterialTheme.colorScheme.primary,
+              contentColor = Color.Unspecified,
+              disabledContentColor = Color.Unspecified,
+              disabledContainerColor = Color.Unspecified)) {
+        Text("Select Time: ${selectedTime.value}", color = MaterialTheme.colorScheme.onPrimary)
```
Comment (Acheline) [inline app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt]: - Suggestion : After changing to "HHmm" in selectedTime.value, change this line of code to :
``
Text("Select Time: ${selectedTime.value.chunked(2).joinToString("h")}")
``

File: app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt [hunk 6]
```diff
@@ -0,0 +1,475 @@
...
+
+          Text(
+              text = "New Event",
+              style = MaterialTheme.typography.titleLarge,
+              textAlign = TextAlign.Center,
+              modifier = Modifier.weight(1f))
+
+          IconButton(
+              onClick = {
+                val sdf = SimpleDateFormat("dd/MM/yyyyHHmm", Locale.getDefault())
```
Comment (Acheline) [inline app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt]: - Nitpick : Make parsing safer and time-zone aware; guard against parse failures explicitly.
- Suggestion : 
``
val sdf = SimpleDateFormat("dd/MM/yyyyHHmm", Locale.getDefault())
sdf.timeZone = java.util.TimeZone.getDefault()
val rawTime = if (time.value.contains("h")) time.value.replace("h","") else time.value
val parsed = runCatching { sdf.parse(date.value + rawTime) }.getOrNull()
val timestamp = parsed?.let { Timestamp(it) } ?: Timestamp.now()
``

File: app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt (hunk 1 above)
Comment (ItsaBox9368) [reply app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt]: Indeed, this is now fixed in a new version

File: app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt (hunk 2 above)
Comment (ItsaBox9368) [reply app/src/main/java/com/swent/mapin/ui/event/AddEventScreen.kt]: Nice thanks for picking up on that

**Review by Acheline — APPROVED** [review]
Looks good to me.
